src/cluster.c

```diff
@@ -1332,16 +1332,18 @@ void addNodeToNodeReply(client *c, clusterNode *node) {
  * not finished their initial sync, in failed state, or are
  * otherwise considered not available to serve read commands. */
 int isNodeAvailable(clusterNode *node) {
+    /* We don't consider PFAIL here because it's not a reliable indicator
+     * for node available and we don't want clients to use it. */
     if (clusterNodeIsFailing(node)) {
         return 0;
     }
-    long long repl_offset = clusterNodeReplOffset(node);
-    if (clusterNodeIsMyself(node)) {
-        /* Nodes do not update their own information
-         * in the cluster node list. */
-        repl_offset = getNodeReplicationOffset(node);
+
+    /* Hide empty replicas in here, from a data-path POV, an empty replica
+     * is not available. */
+    if (getNodeReplicationOffset(node) == 0) {
+        return 0;
     }
```

Comment:
or maybe we can have it in previous code style itself.

```suggestion
    return getNodeReplicationOffset(node) != 0;
```

Reply:
i want to keep the empty replica comment for better reading, and in this way we don't have to worry about touching it again. Since we are already touching it, i prefer the new code style.